### ISSUE TITLE
[HuggingFace][Neuronx] Image retrieval missing `sdk` information

### ISSUE THREAD

**Issue by philschmid** [aws/sagemaker-python-sdk#3899]
**Describe the bug**
The Hugging Face image for `inf2` (neuronx) is not retrieved correctly. 

**To reproduce**

```python
from sagemaker.huggingface.model import HuggingFaceModel

huggingface_model = HuggingFaceModel(
    model_data="s3://mybucket/train",
    transformers_version="4.28",
    role=role,
    pytorch_version="1.13",
    py_version="py38",
)
container = huggingface_model.prepare_container_def("ml.inf2.xlarge", inference_tool="neuronx")
```
returns 
`https://763104351884.dkr.ecr.us-east-1.amazonaws.com/huggingface-pytorch-inference-neuronx:1.13-transformers4.28-neuronx-py38-ubuntu20.04`
which is not a valid tag according to the [release](https://github.com/aws/deep-learning-containers/releases/tag/v1.0-hf-4.28.1-pt-1.13.0-inf-neuronx-sdk2.9.1-py38). Valid tags would be 

```bash
1.13.0-neuronx-py38-sdk2.9.1-ubuntu20.04
1.13.0-transformers4.28.1-neuronx-py38-sdk2.9.1-ubuntu20.04-v1.0-2023-05-08-21-23-21
1.13-transformers4.28-neuronx-py38-sdk2.9.1-ubuntu20.04-v1
1.13-neuronx-py38-sdk2.9.1-ubuntu20.04-v1
1.13-transformers4.28-neuronx-py38-sdk2.9.1-ubuntu20.04
1.13.0-transformers4.28.1-neuronx-py38-sdk2.9.1-ubuntu20.04
1.13.0-transformers4.28.1-neuronx-py38-sdk2.9.1-ubuntu20.04-v1.0
```

The SageMaker retrieval is missing the SDK version or the Container without the SDK version is missing

cc @arjkesh

**Comment by arjkesh** [comment]
@philschmid can confirm that all of the DLC tags have neuron SDK version - we do not have a tag that omits this information. This is also true of the HF PT inference neuron images we released for PT 1.10. Also confirmed on the non-huggingface neuron images. In order to retrieve the information, we will need to provide neuron sdk version while generating the tag

**Comment by philschmid** [comment]
Why was that issue closed?! The mentioned PR clearly says "related to https://github.com/aws/sagemaker-python-sdk/issues/3899
after image is retrieved correctly this PR needs to be merged."

It did not fix this issue! it fixes another issue!
